### DISCUSSION TITLE
Documentation for сontributors

### DISCUSSION THREAD

**Discussion by ltlaitoff** [csandman/chakra-react-select#313] [Ideas]
It would be great to have documentation for contributors because it's not completely clear how to `locally run` the project and how to `test` it

When I find a simple bug, I want to dive into the code and try to fix it, rather than creating a task that will be fixed someday

The documentation should include instructions on how to install the project locally, how to run and debug it. How to properly create PRs, how to commit (conventions), etc

After implementing all this, anyone will be able to easily contribute or fix a bug that bothers them

**Comment by csandman** [comment]
Haha, yeah that's something I've been meaning to do, but haven't gotten around to it. The problem is, I never put the time in to setting up a proper local development environment, so even my approach to developing this package is pretty janky.

I'll see if I can find the time to set up a proper local development environment first, and if I can I'll add some documentation about it.

One thing I was thinking about is that I could potentially copy the testing suite from the original `react-select` package into this one, as that would probably cover most of the integration tests at least. And besides that, I was thinking it might be easiest to develop in a storybook environment, but I'm not set on that.

If you have any ideas or examples for a good local development project setup to follow, I'm all ears!
